test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -1559,6 +1559,102 @@ public async Task TimeoutDuringAuthentication()
                     .With.InnerException.TypeOf<TimeoutException>());
         }
 
+        [Test]
+        public async Task Physical_open_callback_sync()
+        {
+            await using var defaultConn = await OpenConnectionAsync();
+            await using var _ = await CreateTempTable(defaultConn, "ID INTEGER", out var table);
+
+            using var __ = CreateTempPool(ConnectionString, out var connectionString);
+            using var conn = new NpgsqlConnection(connectionString);
+            conn.PhysicalOpenCallback = connector =>
+            {
+                using var cmd = connector.CreateCommand($"INSERT INTO \"{table}\" VALUES(1)");
+                cmd.ExecuteNonQuery();
+                return new ValueTask();
+            };
+
+            conn.Open();
+
+            var rowsCount = (long)(await conn.ExecuteScalarAsync($"SELECT COUNT(*) FROM \"{table}\""))!;
+            Assert.AreEqual(1, rowsCount);
+        }
+
+        [Test]
+        public async Task Physical_open_callback_async()
+        {
+            await using var defaultConn = await OpenConnectionAsync();
+            await using var _ = await CreateTempTable(defaultConn, "ID INTEGER", out var table);
+
+            using var __ = CreateTempPool(ConnectionString, out var connectionString);
+            await using var conn = new NpgsqlConnection(connectionString);
+            conn.PhysicalOpenCallback = async connector =>
+            {
+                using var cmd = connector.CreateCommand($"INSERT INTO \"{table}\" VALUES(1)");
+                await cmd.ExecuteNonQueryAsync();
+            };
+
+            await conn.OpenAsync();
+
+            var rowsCount = (long)(await conn.ExecuteScalarAsync($"SELECT COUNT(*) FROM \"{table}\""))!;
+            Assert.AreEqual(1, rowsCount);
+        }
+
+        [Test]
+        public async Task Physical_open_callback_async_over_sync()
```

Comment:
This is actually sync over async no?

Reply:
We're testing here that the synchronous `Open` throws, if `PhysicalOpenCallback` is async - wouldn't sync over async be the other way arond?

Reply:
Well, this would be sync Open calling the async PhysicalOpenCallback, so sync-over-async, no?

Reply:
OK

Reply:
FWIW if this were async-over-sync (i.e. sync callback being invoked from OpenAsync), it would only mean that OpenAsync wouldn't be 100% asynchronous. This is much less of a big deal than sync-over-async, which easily triggers TP starvation and what-not...